### PR TITLE
feat(e2ei): add e2ei certificate manual enrollment (WPB-309)

### PR DESCRIPTION
**Pull request by mchenani**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-309" title="WPB-309" target="_blank">EpicWPB-309</a>  [AR] E2EI Manually issue certificate
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Add google OAuth and E2EI manual enrolment in debug screen.

### Testing
WIP

#### How to Test

- Against MLS Enabled backend
- Have you account added to the Wire-E2EI OAuth test server
- You should be able to enrol the E2EI Certificate and before that you must authorise OAuth via GoogleOAuth.

### Attachments (Optional)

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1840/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1840/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1840/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1840/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1840/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1840/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1840/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [130](https://10.10.124.134/job/AR-build-pipeline/130/) **failed**.

**Comment by AndroidBob** [comment]
Build [131](https://10.10.124.134/job/AR-build-pipeline/131/) **failed**.

**Comment by AndroidBob** [comment]
Build [133](https://10.10.124.134/job/AR-build-pipeline/133/) **failed**.

**Comment by AndroidBob** [comment]
Build [134](https://10.10.124.134/job/AR-build-pipeline/134/) **failed**.

**Comment by AndroidBob** [comment]
Build [137](https://10.10.124.134/job/AR-build-pipeline/137/) **failed**.

File: app/src/main/kotlin/com/wire/android/feature/OAuthUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    )
+    private var appAuthConfiguration: AppAuthConfiguration = AppAuthConfiguration.Builder().setBrowserMatcher(
+        BrowserAllowList(
+            VersionedBrowserMatcher.CHROME_CUSTOM_TAB, VersionedBrowserMatcher.SAMSUNG_CUSTOM_TAB
+        )
+    ).build()
+
+    init {
+        authorizationService = AuthorizationService(context, appAuthConfiguration)
+    }
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/feature/OAuthUseCase.kt]: why `var`s not `val`s????
and you can merge initialization of `authorizationService`

File: app/src/main/kotlin/com/wire/android/feature/OAuthUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    companion object {
+        const val OAUTH_ACTIVITY_RESULT_KEY = "OAuthActivityResult"
+
+        const val SCOPE_PROFILE = "profile"
+        const val SCOPE_EMAIL = "email"
+        const val SCOPE_OPENID = "openid"
+
+        const val CLIENT_ID = "338888153072-4fep6tn6k16tmcbhg4nt4lr65pv3avgi.apps.googleusercontent.com"
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/feature/OAuthUseCase.kt]: is it save to store it as a const?

**Review by borichellow — APPROVED** [review]
nice one 💪🤟, just a few questions/suggestions :)

**Comment by AndroidBob** [comment]
Build [611](https://10.10.124.134/job/AR-build-pipeline/611/) **failed**.

**Comment by AndroidBob** [comment]
Build [671](https://10.10.124.134/job/AR-build-pipeline/671/) **failed**.

**Comment by AndroidBob** [comment]
Build [672](https://10.10.124.134/job/AR-build-pipeline/672/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7138900)(7138900)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7138900)(7138900)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [786](https://10.10.124.134/job/AR-build-pipeline/786/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7190934)(7190934)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7190934)(7190934)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/feature/OAuthUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        const val CODE_VERIFIER_CHALLENGE_METHOD = "S256"
+        const val MESSAGE_DIGEST_ALGORITHM = "SHA-256"
+        val MESSAGE_DIGEST = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM)
+        const val ENCODING = Base64.URL_SAFE or Base64.NO_PADDING or Base64.NO_WRAP
+
+        const val URL_AUTHORIZATION = "https://accounts.google.com/o/oauth2/auth"
+        const val URL_TOKEN_EXCHANGE = "https://oauth2.googleapis.com/token"
+        const val URL_AUTH_REDIRECT = "com.wire.android.internal.debug:/oauth2redirect"
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/feature/OAuthUseCase.kt]: Should we somehow depend on build config to have different redirect to different builds?

**Review by Garzas — APPROVED** [review]
Looking good 🚀 Just small question

**Comment by AndroidBob** [comment]
Build [930](https://10.10.124.134/job/AR-build-pipeline/930/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7242936)(7242936)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7242936)(7242936)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [930](https://10.10.124.134/job/AR-build-pipeline/930/) **failed**.

**Comment by AndroidBob** [comment]
Build [966](https://10.10.124.134/job/AR-build-pipeline/966/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7254192)(7254192)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7254192)(7254192)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [628](https://10.10.124.134/job/AR-build-pipeline/628/) **failed**.

**Comment by AndroidBob** [comment]
Build [661](https://10.10.124.134/job/AR-build-pipeline/661/) **failed**.

**Comment by AndroidBob** [comment]
Build [935](https://10.10.124.134/job/AR-build-pipeline/935/) **failed**.

**Comment by AndroidBob** [comment]
Build [1397](https://10.10.124.134/job/AR-build-pipeline/1397/) **failed**.

**Comment by AndroidBob** [comment]
Build [1652](https://10.10.124.134/job/AR-build-pipeline/1652/) **failed**.

**Comment by AndroidBob** [comment]
Build [1665](https://10.10.124.134/job/AR-build-pipeline/1665/) **failed**.

**Comment by AndroidBob** [comment]
Build [1667](https://10.10.124.134/job/AR-build-pipeline/1667/) **failed**.

**Comment by AndroidBob** [comment]
Build [1734](https://10.10.124.134/job/AR-build-pipeline/1734/) **failed**.

**Comment by AndroidBob** [comment]
Build [1786](https://10.10.124.134/job/AR-build-pipeline/1786/) **failed**.

**Comment by AndroidBob** [comment]
Build [1787](https://10.10.124.134/job/AR-build-pipeline/1787/) **failed**.

**Comment by AndroidBob** [comment]
Build [1807](https://10.10.124.134/job/AR-build-pipeline/1807/) **failed**.

**Comment by AndroidBob** [comment]
Build [1816](https://10.10.124.134/job/AR-build-pipeline/1816/) **failed**.

**Comment by AndroidBob** [comment]
Build [1818](https://10.10.124.134/job/AR-build-pipeline/1818/) **failed**.

**Comment by AndroidBob** [comment]
Build [1825](https://10.10.124.134/job/AR-build-pipeline/1825/) **failed**.

**Comment by AndroidBob** [comment]
Build [1834](https://10.10.124.134/job/AR-build-pipeline/1834/) **failed**.

**Comment by AndroidBob** [comment]
Build [1835](https://10.10.124.134/job/AR-build-pipeline/1835/) **failed**.

**Comment by AndroidBob** [comment]
Build [1879](https://10.10.124.134/job/AR-build-pipeline/1879/) **failed**.

**Comment by AndroidBob** [comment]
Build [1881](https://10.10.124.134/job/AR-build-pipeline/1881/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8126890)(8126890)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8126890)(8126890)-staging-compat.apk)

**Pull request merged**